django/db/migrations/autodetector.py

```diff
@@ -1134,7 +1134,7 @@ def _generate_added_field(self, app_label, model_name, field_name):
         preserve_default = (
             field.null
             or field.has_default()
-            or field.db_default is not models.NOT_PROVIDED
+            or field.has_db_default()
             or field.many_to_many
             or (field.blank and field.empty_strings_allowed)
             or (isinstance(field, time_fields) and field.auto_now)
```

Comment:
I still haven't opened a PR. The reason is I don't think it would help. If `has_db_default` is true, some code would try to use the `db_default`,  which would raise errors.

I'm looking for alternatives.

Reply:
No problem 👍

Reply:
Instead of hacking around `default` / `db_default`, I just set them to a real expression when creating the object. So there's no need to have a `had_db_default` function.